src/io/flutter/run/Launcher.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run;
+
+import com.intellij.execution.*;
+import com.intellij.execution.configurations.CommandLineState;
+import com.intellij.execution.configurations.RunProfile;
+import com.intellij.execution.configurations.RunProfileState;
+import com.intellij.execution.executors.DefaultDebugExecutor;
+import com.intellij.execution.executors.DefaultRunExecutor;
+import com.intellij.execution.filters.TextConsoleBuilder;
+import com.intellij.execution.filters.TextConsoleBuilderImpl;
+import com.intellij.execution.filters.UrlFilter;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.execution.runners.ExecutionEnvironment;
+import com.intellij.execution.runners.GenericProgramRunner;
+import com.intellij.execution.runners.ProgramRunner;
+import com.intellij.execution.runners.RunContentBuilder;
+import com.intellij.execution.ui.ConsoleView;
+import com.intellij.execution.ui.RunContentDescriptor;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.Separator;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.xdebugger.XDebugProcess;
+import com.intellij.xdebugger.XDebugProcessStarter;
+import com.intellij.xdebugger.XDebugSession;
+import com.intellij.xdebugger.XDebuggerManager;
+import com.jetbrains.lang.dart.analyzer.DartAnalysisServerService;
+import com.jetbrains.lang.dart.ide.runner.DartRelativePathsConsoleFilter;
+import com.jetbrains.lang.dart.util.DartUrlResolver;
+import io.flutter.actions.OpenObservatoryAction;
+import io.flutter.actions.OpenSimulatorAction;
+import io.flutter.dart.DartPlugin;
+import io.flutter.run.daemon.DeviceService;
+import io.flutter.run.daemon.FlutterApp;
+import io.flutter.run.daemon.FlutterDevice;
+import io.flutter.run.daemon.RunMode;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Launches a flutter app, showing it in the console.
+ *
+ * <p>Normally creates a debugging session, which is needed for hot reload.
+ */
+public class Launcher extends CommandLineState {
+  private final @NotNull VirtualFile workDir;
+
+  /**
+   * The file or directory holding the Flutter app's source code.
+   * This determines how the analysis server resolves URI's (for breakpoints, etc).
+   *
+   * <p>If a file, this should be the file containing the main() method.
```

Comment:
I haven't dug into the debugger code enough to know for sure, but my guess is that if it's the wrong directory entirely, the debugger will have trouble translating between package URL's and source filenames, and therefore setting breakpoints won't work.